cmd/zfs/zfs_iter.c

```diff
@@ -213,10 +213,57 @@ zfs_free_sort_columns(zfs_sort_column_t *sc)
 }
 
 int
-zfs_sort_only_by_name(const zfs_sort_column_t *sc)
+zfs_sort_only_by_fast(const zfs_sort_column_t *sc)
 {
-	return (sc != NULL && sc->sc_next == NULL &&
-	    sc->sc_prop == ZFS_PROP_NAME);
+	if (sc == NULL) {
+		/* No sort means we don't need any extra properties */
+		return (1);
+	}
+
+	while (sc != NULL) {
+		switch (sc->sc_prop) {
+		case ZFS_PROP_NAME:
+		case ZFS_PROP_GUID:
+		case ZFS_PROP_CREATETXG:
+		case ZFS_PROP_NUMCLONES:
+		case ZFS_PROP_INCONSISTENT:
+		case ZFS_PROP_REDACTED:
+		case ZFS_PROP_ORIGIN:
+			break;
+		default:
+			return (0);
+		}
+		sc = sc->sc_next;
+	}
+
+	return (1);
+}
+
+int
```

Comment:
Please return `boolean_t` from both of these checks for readability. `B_TRUE` and `B_FALSE` are unambiguous.